mako/exceptions.py

```diff
@@ -235,14 +231,12 @@ def _init(self, trcback):
             else:
                 if new_trcback:
                     try:
-                        # A normal .py file (not a Template)
-                        fp = open(new_trcback[-1][0], "rb")
-                        encoding = util.parse_encoding(fp)
-                        if compat.py3k and not encoding:
-                            encoding = "utf-8"
-                        fp.seek(0)
-                        self.source = fp.read()
-                        fp.close()
+                        with open(new_trcback[-1][0], "rb") as fp:
```

Comment:
the comment `# A normal .py file (not a Template)` should be kept I think

Reply:
hey @CaselIT, Thanks for the review i will revert the comment to this part